internal/fleet/integration_policy/models.go

```diff
@@ -0,0 +1,149 @@
+	model.VarsJson = utils.MapToNormalizedType(utils.Deref(data.Vars), path.Root("vars_json"), diags)
+
+	{
+		newInputs := utils.TransformMapToSlice(data.Inputs, path.Root("input"), diags,
+			func(inputData fleetapi.PackagePolicyInput, meta utils.MapMeta) integrationPolicyInputModel {
+				return integrationPolicyInputModel{
+					InputID:     types.StringValue(meta.Key),
+					Enabled:     types.BoolValue(inputData.Enabled),
+					StreamsJson: utils.MapToNormalizedType(utils.Deref(inputData.Streams), meta.Path.AtName("streams_json"), diags),
+					VarsJson:    utils.MapToNormalizedType(utils.Deref(inputData.Vars), meta.Path.AtName("vars_json"), diags),
+				}
+			})
+		if newInputs == nil {
+			model.Input = types.ListNull(getInputType())
+		} else {
+			oldInputs := utils.ListTypeAs[integrationPolicyInputModel](ctx, model.Input, path.Root("input"), diags)
+			sortInputs(newInputs, oldInputs)
+
+			inputList, d := types.ListValueFrom(ctx, getInputType(), newInputs)
+			diags.Append(d...)
+
+			model.Input = inputList
+		}
+	}
+
+	return diags
+}
+
+func (model integrationPolicyModel) toAPIModel(ctx context.Context, isUpdate bool) (fleetapi.PackagePolicyRequest, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	body := fleetapi.PackagePolicyRequest{
+		Description: model.Description.ValueStringPointer(),
+		Force:       model.Force.ValueBoolPointer(),
+		Name:        model.Name.ValueString(),
+		Namespace:   model.Namespace.ValueStringPointer(),
+		Package: struct {
+			Name    string `json:"name"`
+			Version string `json:"version"`
+		}{
+			Name:    model.IntegrationName.ValueString(),
+			Version: model.IntegrationVersion.ValueString(),
+		},
+		PolicyId: model.AgentPolicyID.ValueString(),
+		Vars:     utils.MapRef(utils.NormalizedTypeToMap[any](model.VarsJson, path.Root("vars_json"), diags)),
+	}
+
+	if isUpdate {
+		body.Id = model.ID.ValueStringPointer()
+	}
+
+	body.Inputs = utils.MapRef(utils.ListTypeToMap(ctx, model.Input, path.Root("input"), diags,
+		func(inputModel integrationPolicyInputModel, meta utils.ListMeta) (string, fleetapi.PackagePolicyRequestInput) {
+			return inputModel.InputID.ValueString(), fleetapi.PackagePolicyRequestInput{
+				Enabled: inputModel.Enabled.ValueBoolPointer(),
+				Streams: utils.MapRef(utils.NormalizedTypeToMap[fleetapi.PackagePolicyRequestInputStream](inputModel.StreamsJson, meta.Path.AtName("streams_json"), diags)),
+				Vars:    utils.MapRef(utils.NormalizedTypeToMap[any](inputModel.VarsJson, meta.Path.AtName("vars_json"), diags)),
+			}
+		}))
+
+	return body, diags
+}
+
+// sortInputs will sort the 'incoming' list of input definitions based on
+// the order of inputs defined in the 'existing' list. Inputs not present in
+// 'existing' will be placed at the end of the list. Inputs are identified by
+// their ID ('input_id'). The 'incoming' slice will be sorted in-place.
+func sortInputs(incoming []integrationPolicyInputModel, existing []integrationPolicyInputModel) {
+	if len(existing) == 0 {
```

Comment:
This is a change from previous that I think addresses a bug that was not noticed before. When importing, existing will be empty/null. And if you have more than 1 import, the order will get randomized (thanks go-maps), so the import test will randomly fail.